src/conn.rs

```diff
@@ -0,0 +1,18 @@
+use crate::extensions::idle::SetReadTimeout;
+
+use std::fmt::{Debug, Formatter};
+use std::io::{Read, Write};
+
+/// Imap connection trait of a read/write stream
+pub trait ImapConnection: Read + Write + SetReadTimeout {}
```

Comment:
We should probably [seal](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) this trait.